src/jcli/file/migrations-man.ts

```diff
@@ -0,0 +1,47 @@
+import { DBClass } from "@/api/mod.ts";
+import { FileChange, FileEntry, listFiles } from "@/jcli/file/files-man.ts";
+
+const BASE_PATH = "./migrations";
+
+async function buildFileChange(
+  entry: FileEntry,
+  existingMigrationHashes: Map<string, string>,
+): Promise<FileChange | undefined> {
+  const entryDigest = await entry.digest();
+
+  if (existingMigrationHashes.has(entry.path)) {
+    if (entryDigest !== existingMigrationHashes.get(entry.path) as string) {
+      return {
+        type: "UPDATED",
+        path: entry.path,
+        digest: entryDigest,
+      };
+    }
+  } else {
+    return { type: "CREATED", path: entry.path, digest: entryDigest };
+  }
+}
+
+export async function* diffMigrations(db: DBClass): AsyncIterable<FileChange> {
+  const existingMigrationHashes = new Map(db.query<[string, string]>(
+    "SELECT path, hash FROM file_hashes WHERE filetype = 'MIGRATIONS'",
+  ));
+
+  const newMigrationPaths = new Set<string>();
+
+  for await (const entry of listFiles(BASE_PATH, ".sql")) {
+    newMigrationPaths.add(entry.path);
+
+    const fileChange = await buildFileChange(entry, existingMigrationHashes);
+
+    if (fileChange !== undefined) {
```

Comment:
为啥要显式判断 undefined 呢？
```suggestion
    if (fileChange) {
```